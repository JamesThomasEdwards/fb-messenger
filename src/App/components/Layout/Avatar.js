import React from "react";
import styled from "styled-components";

const AvatarWrapper = styled.img`
  margin-right: 0.5em;

  ${props =>
    props.size === "small" &&
    `
    clip-path: circle(10px at center);
    width:20px;
  `}
  
  ${props =>
    props.size === "medium" &&
    `
    clip-path: circle(15px at center);
    width:30px;
    height: 30px;
  `}
  
  ${props =>
    props.size === "large" &&
    `
    clip-path: circle(25px at center);
    width: 50px;
  `}
`;

const Avatar = ({ username, size = "medium" }) => (
  <AvatarWrapper
    size={size}
    src={`/images/${username}_lg.jpg`}
    alt={`${username}`}
  />
);

export default Avatar;
