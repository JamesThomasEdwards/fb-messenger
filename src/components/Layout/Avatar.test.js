import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Avatar from "./Avatar";

describe("<Avatar />", () => {
  it("renders Avatar", () => {
    const wrapper = shallow(<Avatar username="alex" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
