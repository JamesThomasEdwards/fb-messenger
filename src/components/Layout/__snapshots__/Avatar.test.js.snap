// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders Avatar 1`] = `
<styled.img
  alt="alex"
  size="medium"
  src="/images/alex_lg.jpg"
/>
`;
